src/BaseApiAbstract.php

```diff
@@ -232,28 +233,33 @@ public function __construct($projectId, ClientInterface $client, $logger = null,
     /**
      * @param string $serviceUrl
      * @param bool $debug
+     * @param HandlerStack|null $handler
      *
      * @return Client
      */
-    protected static function initializeHttpClient($serviceUrl, $debug = false)
+    public static function initializeHttpClient($serviceUrl, $debug = false, $handler = null)
```

Comment:
@PavelLoparev Is it possible to eliminate `$debug` from params since we add `$handler`? Just curious, I'm not familiar with guzzle and `handler`.

Reply:
Default values always smell like a quick workaround instead of refactoring + fixing tests. One more param with default value just adds confidence that developer doesn't want to touch method code and be responsible for it.
Also, this method becomes `public` so we will not be able to change its signature easily in the future.

What about:
1. Create HttpClientConfiguration class (or maybe guzzle has something like this). Thus dev can define all required configuration (an example proxy creds) and pass it into SDK
2. Or maybe allow to set entire preconfigured HTTP client. But in this case it will not use proper user-agent.